edx_user_state_client/tests.py

```diff
@@ -352,6 +384,79 @@ def test_delete_many_last_field(self):
         self.delete_many(user=0, blocks=[0, 1], fields=['a', 'b'])
         self.assertItemsEqual(self.get_many(user=0, blocks=[0, 1]), [])
 
+    def test_get_mod_date(self):
+        start_time = datetime.now()
+        self.set_many(user=0, block_to_state={0: {'a': 'b'}, 1: {'b': 'c'}})
+        end_time = datetime.now()
+
+        mod_dates = self.get_mod_date(user=0, block=0)
+
+        self.assertItemsEqual(mod_dates.state.keys(), ["a"])
+        self.assertGreater(mod_dates.updated, start_time)
+        self.assertLess(mod_dates.updated, end_time)
+
+    def test_get_mod_date_many(self):
+        self.set_many(
+            user=0,
+            block_to_state={0: {'a': 'b'}, 1: {'b': '_', 'c': 'd'}})
+        start_time1 = datetime.now()
+        self.set_many(
+            user=0,
+            block_to_state={1: {'b': 'c', 'c': 'd', 'd': 'e'}})
+        end_time1 = datetime.now()
+
+        mod_dates = list(self.get_mod_date_many(
+            user=0,
+            blocks=[0, 1],
+            fields=["b", "c"]))
+
+        self.assertItemsEqual(
+            [result.block_key for result in mod_dates],
+            [self._block(1)])
+        self.assertItemsEqual(
+            sorted(mod_dates[0].state.keys()),
+            ["b", "c"])
+        self.assertGreater(mod_dates[0].updated, start_time1)
+        self.assertLess(mod_dates[0].updated, end_time1)
+
+        start_time2 = datetime.now()
+        self.set_many(
+            user=0,
+            block_to_state={1: {'a': 'new', 'b': 'c2', 'c': 'd'}})
+        end_time2 = datetime.now()
```

Comment:
I think it would be better to have this as a separate test case that is explicitly about the independence of mod dates between blocks.